src/Aliencube.AzureFunctions.Extensions.OpenApi.Core/Extensions/TypeExtensions.cs

```diff
@@ -361,6 +361,12 @@ public static string GetOpenApiReferenceId(this Type type, bool isDictionary, bo
                 return namingStrategy.GetPropertyName(name, hasSpecifiedName: false);
             }
 
+            if (type.IsGenericType)
+            {
+                return namingStrategy.GetPropertyName(type.Name.Split('`').First(), false) + "_" +
```

Comment:
I think this makes sense. Which one would you think the best to replace backtick with?

1. dot
2. hyphen/dash
3. underscore.

I know you chose underscore. If we choose the other, like dot or hyphen, what will happen?

Reply:
I found a workaround. I created a custom NamingStrategy that did pretty much the same thing:

```
        public class CustomCamelCaseNamingStrategy : CamelCaseNamingStrategy
        {
            protected override string ResolvePropertyName(string name)
            {
                return base.ResolvePropertyName(name).Replace("`", "_");
            }
        }
```

The result creates valid openapi document. I tested with changing "_" to "." and "-" as well, and the document is still valid. So I guess someone just have to choose :)

Reply:
Makes sense.

Reply:
For now, let's use this hard-coded value of underscore. But in a long-run, it may let the developers choose one of three options - dot, hyphen or underscore. Created a new issue, #137